sql_script/MS3_insertions.sql

```diff
@@ -1,41 +1,42 @@
-/*INSERTIONS___
-#Add new inventory types to the inventory_type table*/
-INSERT
-INTO inventory_type(type_name, height, width, length)
-VALUES (new_type_name, new_height, new_width, new_length);#impact on dimensions table?
-
-/*Add a new recipient to the recipient table*/
-INSERT
-INTO recipient(id, street_address, postal_code)
+/*Add new inventory types to the inventory_type table 
+-> NOTE: only allows insertion when dimensions already exist in dimensions table*/
+INSERT INTO inventory_type(type_name, height, width, length)
+VALUES (new_type_name, new_height,new_width,new_length);
+
+/*Add a new recipient to the recipient table 
+-> NOTE: only allows insertion if postal code already exists in region*/
+INSERT INTO recipient(id, street_address, postal_code)
 VALUES (new_id, new_street_address, new_postal_code);
 
-/*Add a new producer to the producer table*/
+/*Add a new producer to the producer table
+-> NOTE: only works when new_postal_code already exists in region table*/
```

Comment:
Implement check in back-end and add tuple to region if it doesn't already exist.

Reply:
If you want to add the postal code data (and not just fail if it doesn't already exist), you'll want to run this query beforehand:

```sql
INSERT INTO region
VALUES (new_postal_code, new_municipality, new_province_or_territory)
ON CONFLICT (postal_code) DO NOTHING;
```

The value will be substituted for $1, $2, $3 in the code. $2 and $3 will need to be provided by the user as well.